code/modules/mob/living/carbon/carbon.dm

```diff
@@ -269,14 +269,90 @@ mob/living
 						)
 			// BEGIN HUGCODE - N3X
 			else
-				playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+				if(M.zone_sel.selecting == "r_arm" || M.zone_sel.selecting == "l_arm")
+					playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+					M.visible_message(\
+					"<span class='notice'>[M] pats [src] on their shoulder.</span>",\
+					"<span class='notice'>You pat [src] on their shoulder.</span>",\
+					)
+					
+					
 				if(M.zone_sel.selecting == "head")
+					if (src.get_species() == "Tajaran"||src.get_species() == "Vulpkanin"||src.get_species() == "Farwa"||src.get_species() == "Wolpin")
```

Comment:
how this compiles is byond me, but it'll definately runtime if a xeno is help-intent-attacked by a human